### PR TITLE
MGDOBR-171: manager provides kafka configuration to shard for the internal topic communication

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-171](https://issues.redhat.com/browse/MGDOBR-171) 

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by r00ta** [comment]
I will take a look at the CI tomorrow morning

**Review by robpblake — COMMENTED** [review]
@r00ta I think this is great work as always. My only question over whether or not we needed to go so far as to extract the Kafka configuration to a `Secret`? What was your thoughts there?

I know there are more changes here than in #271 , but I really think this helps to enforce a clean separation between Manager and Shard. The Manager owns the decisions around provisioning, the Shard simply gets told what to do.

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/BridgeIngressServiceImpl.java [hunk 1]
```diff
@@ -144,4 +143,36 @@ public void deleteBridgeIngress(BridgeDTO bridgeDTO) {
             LOGGER.warn("BridgeIngress '{}' not deleted", bridgeDTO);
         }
     }
+
+    @Override
+    public void createOrUpdateBridgeIngressSecret(BridgeIngress bridgeIngress, BridgeDTO bridgeDTO) {
```
[the remark this answers is not in the document]
Comment (r00ta) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/BridgeIngressServiceImpl.java:148]: @robpblake it's best practice to store confidential info like credentials/passwords in Secrets. In such way it's possible to limit the access to Secrets using RBAC policies (usually very few roles can access Secrets, and many can access ConfigMaps instead). 
In addition to that, Kubernetes tries to avoid logging secrets and credentials in general

[the remark this answers is not in the document]
Comment (robpblake) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/BridgeIngressServiceImpl.java:148]: @r00ta I'm totally with you on best practice, but I'd say just keep an eye on scope of your PRs. A lot of the changes here are related to your decision to move from `env` vars to a `Secret`, and that adds additional overhead on reviewers to understand that. Two smaller PRs would help the team to review, particularly with our different individual levels of knowledge on K8S.

I'm not suggesting we remove the usage of a `Secret` as you've already done the work and it's great quality. Just something for future consideration.

[the remark this answers is not in the document]
Comment (lucamolteni) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/BridgeIngressServiceImpl.java:148]: +1

File: shard-operator-integration-tests/src/test/java/com/redhat/service/bridge/shard/operator/cucumber/steps/BridgeIngressSteps.java [hunk 2]
```diff
@@ -26,9 +29,22 @@ public BridgeIngressSteps(Context context) {
     }
 
     @When("^deploy BridgeIngress:$")
-    public void deployBridgeIngress(String bridgeIngressYaml) {
+    public BridgeIngress deployBridgeIngress(String bridgeIngressYaml) {
```
Comment (sutaakar) [inline shard-operator-integration-tests/src/test/java/com/redhat/service/bridge/shard/operator/cucumber/steps/BridgeIngressSteps.java:32]: If this step is replaced by the following one then you can merge them, no need to keep here a step which is not used.

**Comment by sutaakar** [comment]
The code looks good, will try it in cluster

**Review by sutaakar — APPROVED** [review]
Seems to work well

File: shard-operator/src/test/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceTest.java [hunk 3]
```diff
@@ -122,6 +133,7 @@ public void testNotifyBridgeStatusChange() throws InterruptedException {
     }
 
     @Test
+    @WithPrometheus
```
Comment (danielezonca) [inline shard-operator/src/test/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceTest.java:136]: Why this change?

**Comment by sutaakar** [comment]
Minor adjustments done in https://github.com/r00ta/sandbox/pull/2

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/BridgeExecutorServiceImpl.java [hunk 4]
```diff
@@ -87,15 +88,6 @@ public Deployment fetchOrCreateBridgeExecutorDeployment(BridgeExecutor bridgeExe
 
         // TODO: All the Executor applications will push events to the same kafka cluster under the same kafka topic. This configuration will have to be specified by the manager for each Bridge instance: https://issues.redhat.com/browse/MGDOBR-123
```
Comment (danielezonca) [inline shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/BridgeExecutorServiceImpl.java:89]: This TODO is wrong, I think the ticket is now MGDOBR-58

**Review by danielezonca — APPROVED** [review]

**Comment by r00ta** [comment]
> Minor adjustments done in [r00ta#2](https://github.com/r00ta/sandbox/pull/2) @r00ta Can you please check?

Awesome, thank you @sutaakar !

File: shard-operator/src/test/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceTest.java (hunk 3 above)
Comment (r00ta) [reply shard-operator/src/test/java/com/redhat/service/bridge/shard/operator/ManagerSyncServiceTest.java:136]: Because I found out that this is currently working just because of luck :) If you execute the test standalone it does not work because Prometheus is not installed on the mock.
This is currently working because the test `testBridgesAreDeployed` is annotated with `WithPrometheus` and is executed *before* this test: all the tests that are executed after  `testBridgesAreDeployed` have prometheus installed as well.

File: shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/BridgeExecutorServiceImpl.java (hunk 4 above)
Comment (r00ta) [reply shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/BridgeExecutorServiceImpl.java]: @danielezonca actually I think this TODO has to be removed as the shard does not need anymore to know such details once this PR will be merged. 
Removed
